pkg/sql/lexer.go

```diff
@@ -99,7 +99,8 @@ func (l *lexer) lexIdentOrKeyword(lval *sqlSymType) int {
 	// lexToken ensures that the first rune is a letter.
 	r := l.next()
 	for {
-		for unicode.IsLetter(r) || unicode.IsNumber(r) || r == '_' {
+		// model IDENT may be like: a_data_scientist/regressors:v0.2/MyDNNRegressor
+		for unicode.IsLetter(r) || unicode.IsNumber(r) || r == '_' || r == '/' || r == ':' {
```

Comment:
For the record, SQL syntax usually doesn't accept `/` or `:` being part of the identifier. Take MySQL for example, we can create table named `temp1`, but not `tem/p1` or `tem:p1`.

```SQL
mysql> create table temp1 (a integer);
Query OK, 0 rows affected (0.02 sec)

mysql> create table tem/p1 (a integer);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '/p1 (a integer)' at line 1
mysql> create table tem:p1 (a integer);
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ':p1 (a integer)' at line 1
```

However, `repo/image:tag` is the native syntax in Docker, so I think it is worthwhile to keep them consistent.

https://github.com/sql-machine-learning/sqlflow/pull/1042/files#r337120122 also mentioned this issue.

Reply:
Agree with keep the same naming with Docker image.